examples/nlp/gpt/conf/gpt_self_rewarding.yaml

```diff
@@ -0,0 +1,281 @@
+defaults:
+  - optional tp_overlap@model.ub_tp_comm_overlap_cfg:
+
+trainer:
+  num_nodes: 8
+  devices: 8
+  accelerator: gpu
+  precision: bf16
+  
+  # self-rewarding specific args
+  self_rewarding:
+    max_iterations: 3   # the number of iterations to conduct self-rewarding for, with the reference policy updated on each iteration
+    max_epochs: 1       # the number of epochs to use per iteration
+    max_steps: -1
+    val_check_interval: 0.1
+    save_interval: 100
+    limit_train_batches: 1.0
+
+    # how many GBS we loop over
+    limit_val_batches: 1.0
+    gradient_clip_val: 1.0
+
+    # Accelerate training times by accelerating inference stage using TRTLLM
+    trt_llm:
+      enable: False
+      # reshard is not supported in Self-Rewarding
+
+      # TRTLLM preallocates activation memory according to the number of input tokens
+      # By default, assume the max input length is half of the model sequence length
+      max_input_len: ${int_div:${model.encoder_seq_length}, 2}
+      generation_batch_size: ${model.spin.rollout_micro_batch_size}
+
+      model_type: gptnext
```

Comment:
Done, but I left it as `gptnext` as primary use case will be Nemotron. PPO should change to follow suit

Reply:
Ok, I'll point @gshennvm to this thread so he can look into PPO

Reply:
thanks -- the reason behind llama is because we want to switch the nemotron configs to use llama instead of gptnext and unify things. but since it's not ready yet I'm ok with gptnext for now